app/models/foreman_tasks/task.rb

```diff
@@ -36,6 +36,7 @@ def check_permissions_after_save
     end
 
     scoped_search :on => :id, :complete_value => false
+    scoped_search :on => :action, :complete_value => false
```

Comment:
I see we are adding yet another implicit field to search. IMHO it would be better to be more explicit, it will make the query simpler and run faster. You can do it easily by adding `:only_explicit => true`

Reply:
For this particular, I would probably leave this in the implicit search TBH